### DISCUSSION TITLE
backpack/calander - Start and End Date not showing datetime.

### DISCUSSION THREAD

**Discussion by NeguinhoAdmin** [Laravel-Backpack/community-forum#1023] [General] [answered]
By selecting Calendar specific timeslot it should appear on insert form. it seem not getting selected date/time. Can someone guide if I am doing something wrong.

also look at my schema where I have multiple business entities while form must capable of having both in calander.

<img width="901" alt="image" src="https://github.com/Laravel-Backpack/community-forum/assets/136070103/63a4f2f1-b6c0-4b3a-be8b-2a4477976779">

<img width="433" alt="image" src="https://github.com/Laravel-Backpack/community-forum/assets/136070103/59df366a-5d89-4d5c-966b-667480e7f9bd">

`
class MOTBookingCrudController extends CrudController
{
    use \Backpack\CRUD\app\Http\Controllers\Operations\ListOperation;
    use \Backpack\CRUD\app\Http\Controllers\Operations\CreateOperation;
    use \Backpack\CRUD\app\Http\Controllers\Operations\UpdateOperation;
    use \Backpack\CRUD\app\Http\Controllers\Operations\DeleteOperation;
    use \Backpack\CRUD\app\Http\Controllers\Operations\ShowOperation;
    use \Backpack\CalendarOperation\CalendarOperation;

    public function setup()
    {
        CRUD::setModel(\App\Models\MOTBooking::class);
        CRUD::setRoute(config('backpack.base.route_prefix') . '/mot-booking');
        CRUD::setEntityNameStrings('MOT Booking', 'MOT Bookings');
    }

    public function setupCalendarOperation()
    {
        // Set the initial view
        CRUD::setOperationSetting('initial-view', 'dayGridMonth');

        // Set the available views
        CRUD::setOperationSetting('views', ['dayGridMonth', 'timeGridWeek', 'timeGridDay']);

        // Are the calendar events editable?
        CRUD::setOperationSetting('editable', true);

        // Set the calendar default colors
        CRUD::setOperationSetting('background_color', fn ($event) => $event->active ? 'green' : 'red');

        // same principle applies to text color
        CRUD::setOperationSetting('text_color', fn ($event) => $event->active ? 'white' : 'black');

        // CalendarOperation will inject a javascript into your create and
        // update views to handle the date fields as we do on our demo.
        CRUD::setOperationSetting('with-javascript-widget', true);
    }

    protected function getCalendarFieldsMap(): array
    {
        return [
            'title' => 'title',
            'start' => 'start',
            'end' => 'end',
            'background_color' => 'background_color',
            'text_color' => 'text_color',
            'all_day' => 'all_day',
        ];
    }

    protected function setupListOperation()
    {
        CRUD::setFromDb();
    }

    protected function setupCreateOperation()
    {
        CRUD::addField([
            'name' => 'status',
            'label' => 'Status',
            'type' => 'enum',
            'options' => ['pending', 'available', 'completed', 'cancelled'],
            'default' => 'pending',
            'tab' => 'General',
        ]);
        CRUD::setValidation(MOTBookingRequest::class);

        CRUD::setFromDb();

        CRUD::addField([
            'name' => 'branch_id',
            'label' => 'Branch',
            'type' => 'select2',
            'entity' => 'branch',
            'attribute' => 'name',
            'model' => Branch::class,
            'tab' => 'General',
        ]);
    }

    protected function setupUpdateOperation()
    {
        $this->setupCreateOperation();
    }
}
`
`        Schema::create('mot_bookings', function (Blueprint $table) {
            $table->id();
            $table->unsignedBigInteger('branch_id')->nullable(false);
            $table->foreign('branch_id')->references('id')->on('branches')->onDelete('restrict');
            $table->string('title')->nullable()->default('MOT Booking');
            $table->dateTime('date_of_appointment')->default(now());
            $table->dateTime('start')->nullable();
            $table->dateTime('end')->nullable();
            $table->string('vehicle_registration');
            $table->string('vehicle_chassis');
            $table->string('vehicle_color');
            $table->boolean('all_day')->nullable(true)->default(false);
            $table->string('customer_name');
            $table->string('customer_contact');
            $table->string('customer_email');
            $table->enum('status', ['pending', 'available', 'completed', 'cancelled'])->default('pending');
            $table->text('notes')->nullable(true);
            $table->string('background_color')->nullable();
            $table->string('text_color')->nullable();
            $table->timestamps();
        });`

**Comment by jcastroa87** [comment, marked as answer]
Hello @NeguinhoAdmin 

You need to get the values and add it to the field, can see our [demo](https://github.com/Laravel-Backpack/demo/blob/main/app/Http/Controllers/Admin/MeetingCrudController.php) for an example:

```
protected function setupCreateOperation(): void
    {
        ...

        $start = request()->has('start') ? Carbon::parse(request('start')) : null;
        $end = request()->has('end') ? Carbon::parse(request('end')) : null;

       ...

        CRUD::field('start')
            ->type('datetime')
            ->wrapper(['class' => 'form-group col-md-6'])
            ->value($start);

        CRUD::field('end')
            ->type('datetime')
            ->wrapper(['class' => 'form-group col-md-6'])
            ->value($end);
    }
```

Cheers.
